### PR TITLE
Replace copy_memory with copy_from_slice + Fix ImageDecoder::load_rect

### PR DESCRIPTION
**Pull request by kngwyu**
I think it's make our code a bit simpler.

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -267,9 +267,8 @@ pub trait ImageDecoder: Sized {
             {
                 let from = &tmp[x as usize * bpp..width as usize * bpp];
 
-                let to = &mut buf[i * width as usize * bpp..width as usize * bpp];
-
-                ::copy_memory(from, to);
+                let start = i * width as usize;
+                buf[start..start + from.len()].copy_from_slice(from);
```
Comment (philipc) [inline src/image.rs]: It looks like you're trying to fix a bug here at the same time, but it still looks wrong to me. I haven't tested, but I would expect it to be:
```rust
let from = &tmp[x as usize * bpp..][..width as usize * bpp];
let to = &mut buf[i * width as usize * bpp..][..width as usize * bpp];
to.copy_from_slice(from);
```

Comment (kngwyu) [reply src/image.rs]: Your code looks correct.
Thanks.

**Comment by kngwyu** [comment]
Since `load_rect` had additional bugs other than @philipc commented, I added a test for it and fixed it.
In addition, how about changing its args from `(&mut self, x: u32, y: u32, length: u32, width: u32)` to `(&mut self, x: u32, y: u32, width: u32, length: u32)`, which is the same order as `GenericImage::sub_image`?
This is a breaking change, but I think not so many people will be affected by it, since this method is currently completely broken.

**Comment by fintelia** [comment]
I'm currently working on more extreme breaking changes to this functionality in #819, so I'd advocate for holding off on changing the argument order here. Otherwise, I'm in favor of making the method slightly less buggy.

**Comment by kngwyu** [comment]
Closed due to #819 merged

**Pull request closed without merging**